server/src/main/java/app/user/UserController.java

```diff
@@ -0,0 +1,43 @@
+package app.user;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserService userService;
+
+    @GetMapping("/{userId}/idprofile")
+    public ResponseEntity<UserProfileResponse> getUserProfile(@PathVariable Integer userId, Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        if (!userDetails.getAuthorities().stream()
+                .anyMatch(auth -> auth.getAuthority().equals("ADMIN") || auth.getAuthority().equals("USER"))) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();

```

Comment:
to jest logika serwisu (biznesowa) . Należy w serwisie utworzyć metodę UserService::GetUserProfile

sam w sobie kontroller powinien mieć kilka linij max : wywołanie serswisu oraz mapowanie 

Po co tak się robi ? Bo gdybyśmy chcieli reużyć tej metody to byśmy nie mogli bo ona jest w kontrollerze a kontroller nie da się łatwo wywołać w innym miesjcu w kodzie w javie, dlatego cała logika jest przenoszona do serwisów